phobos/smurf/sensors.py

```diff
@@ -70,8 +71,9 @@ def __init__(
         self.max_distance = max_distance
         self.opening_height = opening_height
         self.vertical_offset = vertical_offset
-        if self.link is None:
-            raise ValueError("Link used for sensor" + name + "does not exist!")
+        if not isinstance(link, representation.Link):
```

Comment:
if link is str you could try to get it from robot.get_link() at the other occurences as well of course. But I think only the name is needed here anyways isn't it so string would be fine.
Probably the sensor/multisensor base class can already take care of getting the representation.Link from robot. ;)

Reply:
Something similar is implemented in some other sensors. I think we implemented think to fix an error. i wrote a method for the base classes which tries to retrieve the missing link.